.github/workflows/test-psalm.yml

```diff
@@ -0,0 +1,72 @@
+name: Psalm
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - '4.*'
+    paths:
+      - '**.php'
+      - 'composer.*'
+      - 'psalm*'
+      - '.github/workflows/test-psalm.yml'
+  push:
+    branches:
+      - develop
+      - '4.*'
+    paths:
+      - '**.php'
+      - 'composer.*'
+      - 'psalm*'
+      - '.github/workflows/test-psalm.yml'
+
+jobs:
+  build:
+    name: Psalm Analysis
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
+          tools: phpstan, phpunit
+          extensions: intl, json, mbstring, xml, oci8
+          coverage: none
+        env:
+          COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get composer cache directory
+        run: echo "COMPOSER_CACHE_FILES_DIR=$(composer config cache-files-dir)" >> $GITHUB_ENV
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.COMPOSER_CACHE_FILES_DIR }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
```

Comment:
Every run of this action would not have a committed composer.lock, so `hashFiles('**/composer.lock')` will always be empty

Reply:
This is from DevKit's template. It's a workaround to support both projects (with lock file) and libraries (no lock file) with the same template. I'm fine adjusting it for the framework repo specifically, but if we do ever move to a central workflow repo this will be a good maneuver to keep.

Reply:
This will be replaced with the reusable one. So I don't adjust.